Use early exporter (#25)
yaronf/draft-tls-attestation: draft-fossati-tls-attestation.md

- TLS server is the attester, and
- TLS client and server mutually attest towards each other. 

> We will show the message exchanges of the first two cases in sub-sections
> below. The last case will be described in future drafts.

Reviewer: We can be more precise:
```suggestion
We will show the message exchanges of the first two cases in {{attester-cli}} and {{attester-srv}}.
The last case is obtained by composing the first two.
```

This needs adding anchors to the referenced sections like this:
```
## TLS Client as Attester {#attester-cli}
```
and
```
## TLS Server as Attester {#attester-srv}
```

## TLS Client as Attester 

In this use case, the TLS server (acting as a relying party) challenges the TLS
client (as the attester) to provide evidence. The TLS client is the attester
and the TLS server acts as a relying party. The TLS server needs to provide a
nonce in the EncryptedExtensions message to the TLS client so that the
attestation service can feed the nonce into the generation of the evidence. The
TLS server, when receiving the evidence, will have to contact the verifier

> (which is not shown in the diagram).

Reviewer 2: The 2nd sentence of this paragraph duplicates the first one.

~~~~
{: #figure-tls-binder title="Format of TLS channel binder."}

- Nonce is the value provided as a challenge by the relying party.

> - The identity key public fingerprint (ik_pub_fingerprint) is a hash of the

Reviewer: I reckon _identity key public fingerprint_ can be trimmed down to _identity key fingerprint_ without loss of precision -- and with a small but sensible increase in readability :-)

```suggestion
- The identity key fingerprint (ik_pub_fingerprint) is a hash of the
```

  Subject Public Key Info from the leaf X.509 certificate transmitted in
  the handshake.
- The channel binder (channel_binder) is a value obtained from the early

>   exporter mechanism offered by the TLS implementation (see section 7.5 of
>   {{RFC8446}}). This Early Exporter Value (EEV) must be obtained immediately

Reviewer: ```suggestion
  exporter mechanism offered by the TLS implementation ({{Section 7.5 of
  RFC8446}}). This Early Exporter Value (EEV) must be obtained immediately
```

  following the ServerHello message, using 'attestation-binder' as the label

>   and an empty context.

Reviewer 2: I think we should be more formal here, using the notation of RFC 8446, Sec. 7.5.

Author: Alright, I've tweaked this bulletpoint a bit (thanks for the suggestion, I realised I hadn't nailed down the binder length) and added a figure to show how the call should look like. Let me know if that seems in line with what you expected!

Reviewer 2: @ionut-arm Yep, this is much better. I just think that the context is an empty string (""), not an empty list ([]).
